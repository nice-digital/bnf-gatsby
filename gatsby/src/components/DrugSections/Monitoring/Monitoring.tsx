import { type FC } from "react";

import {
	type FeedMonitoringPot,
	type FeedMonitoringPotContent,
} from "@nice-digital/gatsby-source-bnf";

import { type QueryResult, type WithSlug, type WithSlugDeep } from "@/utils";

import { PotSection } from "../PotSection/PotSection";

export type MonitoringProps = QueryResult<
	WithSlugDeep<WithSlug<FeedMonitoringPot>, FeedMonitoringPotContent>
>;

export const Monitoring: FC<MonitoringProps> = (props) => {
	return (
		<PotSection {...props}>
			{({
				contentForPrefix,
				content: {
					contentFor,
					slug,
					monitoringOfPatientParameters,
					patientMonitoringProgrammes,
					therapeuticDrugMonitoring,
				},
			}) => {
				return (
					<>
						{therapeuticDrugMonitoring && (
							<section aria-labelledby={`therapeutic-drug-${slug}`}>
								<h4 id={`therapeutic-drug-${slug}`}>
									Therapeutic drug monitoring
									<span
										className="visually-hidden"
										dangerouslySetInnerHTML={{
											__html: ` ${contentForPrefix} ${contentFor}`,
										}}
									/>
								</h4>
								<div
									dangerouslySetInnerHTML={{
										__html: therapeuticDrugMonitoring,
									}}
								/>
							</section>
						)}

						{monitoringOfPatientParameters && (
							<section aria-labelledby={`patient-parameters-${slug}`}>
								<h4 id={`patient-parameters-${slug}`}>
									Monitoring of patient parameters
									<span
										className="visually-hidden"
										dangerouslySetInnerHTML={{
											__html: ` ${contentForPrefix} ${contentFor}`,
										}}
									/>
								</h4>
								<div
									dangerouslySetInnerHTML={{
										__html: monitoringOfPatientParameters,
									}}
								/>
							</section>
						)}

						{patientMonitoringProgrammes && (
							<section aria-labelledby={`patient-programmes-${slug}`}>
								<h4 id={`patient-programmes-${slug}`}>
									Patient monitoring programmes
									<span
										className="visually-hidden"
										dangerouslySetInnerHTML={{
											__html: ` ${contentForPrefix} ${contentFor}`,
										}}
									/>
								</h4>
								<div
									dangerouslySetInnerHTML={{
										__html: patientMonitoringProgrammes,
									}}
								/>
							</section>
						)}
					</>
				);
			}}
		</PotSection>
	);
};
