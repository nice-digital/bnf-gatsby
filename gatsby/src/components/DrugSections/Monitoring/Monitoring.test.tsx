import { render, screen } from "@testing-library/react";

import { Monitoring, type MonitoringProps } from "./Monitoring";

describe("Monitoring", () => {
	const minimumProps: MonitoringProps = {
		potName: "Monitoring requirements",
		slug: "monitoring-requirements",
		drugClassContent: [],
		drugContent: null,
		prepContent: [],
	};

	describe("outer section", () => {
		it("should match snapshot", () => {
			render(
				<Monitoring
					{...minimumProps}
					drugClassContent={[
						{
							contentFor: "corticosteroids",
							patientMonitoringProgrammes:
								"<p>Some patient monitoring for all corticosteroids</p>",
							monitoringOfPatientParameters: null,
							therapeuticDrugMonitoring: null,
						},
						{
							contentFor: "opioids",
							monitoringOfPatientParameters:
								"<p>Patient parameters for all opioids</p>",
							therapeuticDrugMonitoring: null,
							patientMonitoringProgrammes: null,
						},
					]}
					drugContent={{
						contentFor: "co-codamol",
						therapeuticDrugMonitoring: "<p>Therapeutic for co-codamol</p>",
						monitoringOfPatientParameters: null,
						patientMonitoringProgrammes: null,
					}}
					prepContent={[
						{
							contentFor: "SolpadolÂ® caplets",
							patientMonitoringProgrammes:
								"<p>Patient monitoring for a prep</p>",
							therapeuticDrugMonitoring: null,
							monitoringOfPatientParameters: null,
						},
					]}
				/>
			);

			expect(
				screen.getByRole("region", { name: "Monitoring requirements" })
			).toMatchSnapshot();
		});

		it("should render section with accessible name", () => {
			render(<Monitoring {...minimumProps} />);

			expect(
				screen.getByRole("region", { name: "Monitoring requirements" })
			).toBeInTheDocument();
		});

		it("should render heading 2 with section name", () => {
			render(<Monitoring {...minimumProps} />);

			expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
				"Monitoring requirements"
			);
		});
	});

	describe.each([
		["Therapeutic drug monitoring", "therapeuticDrugMonitoring"],
		["Monitoring of patient parameters", "monitoringOfPatientParameters"],
		["Patient monitoring programmes", "patientMonitoringProgrammes"],
	])("%s section", (heading, propertyName) => {
		const props: MonitoringProps = {
			...minimumProps,
			drugContent: {
				contentFor: "furosemide",
				monitoringOfPatientParameters: null,
				patientMonitoringProgrammes: null,
				therapeuticDrugMonitoring: null,
				[propertyName]: "<p>Monitor electrolytes during treatment.</p>",
			},
		};

		it(`should render section for ${heading} with accessible name`, () => {
			render(<Monitoring {...props} />);
			expect(
				screen.getByRole("region", { name: `${heading} For furosemide` })
			).toBeInTheDocument();
		});

		it(`should render heading 4 for ${heading}`, () => {
			render(<Monitoring {...props} />);
			expect(screen.getByRole("heading", { level: 4 })).toHaveTextContent(
				`${heading} For furosemide`
			);
		});

		it(`should render ${heading} content as HTML`, () => {
			render(<Monitoring {...props} />);
			expect(
				screen.getByText("Monitor electrolytes during treatment.")
			).toHaveProperty("tagName", "P");
		});
	});
});
