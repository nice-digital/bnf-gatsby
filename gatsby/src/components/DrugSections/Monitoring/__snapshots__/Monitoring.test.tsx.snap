// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitoring outer section should match snapshot 1`] = `
<section
  aria-labelledby="monitoring-requirements"
  class="section"
>
  <h2
    id="monitoring-requirements"
  >
    Monitoring requirements
  </h2>
  <section
    aria-labelledby="monitoring-requirements-corticosteroids"
  >
    <h3
      class=""
      id="monitoring-requirements-corticosteroids"
    >
      <span
        class="visually-hidden"
      >
        Monitoring requirements 
      </span>
      <span>
        For all corticosteroids
      </span>
    </h3>
    <section
      aria-labelledby="patient-programmes-corticosteroids"
    >
      <h4
        id="patient-programmes-corticosteroids"
      >
        Patient monitoring programmes
        <span
          class="visually-hidden"
        >
           For all corticosteroids
        </span>
      </h4>
      <div>
        <p>
          Some patient monitoring for all corticosteroids
        </p>
      </div>
    </section>
  </section>
  <section
    aria-labelledby="monitoring-requirements-opioids"
  >
    <h3
      class=""
      id="monitoring-requirements-opioids"
    >
      <span
        class="visually-hidden"
      >
        Monitoring requirements 
      </span>
      <span>
        For all opioids
      </span>
    </h3>
    <section
      aria-labelledby="patient-parameters-opioids"
    >
      <h4
        id="patient-parameters-opioids"
      >
        Monitoring of patient parameters
        <span
          class="visually-hidden"
        >
           For all opioids
        </span>
      </h4>
      <div>
        <p>
          Patient parameters for all opioids
        </p>
      </div>
    </section>
  </section>
  <section
    aria-labelledby="monitoring-requirements-co-codamol"
  >
    <h3
      class=""
      id="monitoring-requirements-co-codamol"
    >
      <span
        class="visually-hidden"
      >
        Monitoring requirements 
      </span>
      <span>
        For co-codamol
      </span>
    </h3>
    <section
      aria-labelledby="therapeutic-drug-co-codamol"
    >
      <h4
        id="therapeutic-drug-co-codamol"
      >
        Therapeutic drug monitoring
        <span
          class="visually-hidden"
        >
           For co-codamol
        </span>
      </h4>
      <div>
        <p>
          Therapeutic for co-codamol
        </p>
      </div>
    </section>
  </section>
  <section
    aria-labelledby="monitoring-requirements-Solpadol(r)-caplets"
  >
    <h3
      class=""
      id="monitoring-requirements-Solpadol(r)-caplets"
    >
      <span
        class="visually-hidden"
      >
        Monitoring requirements 
      </span>
      <span>
        For Solpadol® caplets
      </span>
    </h3>
    <section
      aria-labelledby="patient-programmes-Solpadol(r)-caplets"
    >
      <h4
        id="patient-programmes-Solpadol(r)-caplets"
      >
        Patient monitoring programmes
        <span
          class="visually-hidden"
        >
           For Solpadol® caplets
        </span>
      </h4>
      <div>
        <p>
          Patient monitoring for a prep
        </p>
      </div>
    </section>
  </section>
</section>
`;
